src/function/scalar/string/hex.cpp

```diff
@@ -0,0 +1,217 @@
+			uint8_t byte = (uint8_t)(lower >> (offset - 8));
+
+			// at least one byte space
+			if (byte == 0 && !seen_non_zero && offset > 8) {
+				continue;
+			}
+			seen_non_zero = true;
+			memcpy(ptr, &StringUtil::BYTE_TO_HEXS[static_cast<idx_t>(byte) * 2], 2);
+			ptr += 2;
+			buffer_size += 2;
+		}
+
+		// Allocate empty space
+		auto target = StringVector::EmptyString(result, buffer_size);
+		auto output = target.GetDataWriteable();
+		memcpy(output, buffer, buffer_size);
+
+		target.Finalize();
+		return target;
+	}
+};
+
+static void ToHexFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	auto &input = args.data[0];
+	idx_t count = args.size();
+
+	switch (input.GetType().InternalType()) {
+	case PhysicalType::VARCHAR:
+		UnaryExecutor::ExecuteString<string_t, string_t, HexStrOperator>(input, result, count);
+		break;
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		UnaryExecutor::ExecuteString<int8_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::INT16:
+		UnaryExecutor::ExecuteString<int16_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::INT32:
+		UnaryExecutor::ExecuteString<int32_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::INT64:
+		UnaryExecutor::ExecuteString<int64_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::INT128:
+		UnaryExecutor::ExecuteString<hugeint_t, string_t, HexHugeIntOperator>(input, result, count);
+		break;
+	case PhysicalType::UINT8:
+		UnaryExecutor::ExecuteString<uint8_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::UINT16:
+		UnaryExecutor::ExecuteString<uint16_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::UINT32:
+		UnaryExecutor::ExecuteString<uint32_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	case PhysicalType::UINT64:
+		UnaryExecutor::ExecuteString<uint64_t, string_t, HexIntegralOperator>(input, result, count);
+		break;
+	// case PhysicalType::FLOAT:
+	//	UnaryExecutor::ExecuteString<float, string_t, HexIntegralOperator<false>>(input, result, count);
+	//	break;
+	// case PhysicalType::DOUBLE:
+	//	UnaryExecutor::ExecuteString<double, string_t, HexIntegralOperator<false>>(input, result, count);
+	//	break;
+	default:
+		throw NotImplementedException("Specifier type not implemented");
+	}
+}
+
+static void FromHexFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	D_ASSERT(args.data[0].GetType().InternalType() == PhysicalType::VARCHAR);
+	auto &input = args.data[0];
+	idx_t count = args.size();
+
+	UnaryExecutor::ExecuteString<string_t, string_t, FromHexOperator>(input, result, count);
+}
+
+void HexFun::RegisterFunction(BuiltinFunctions &set) {
+	ScalarFunctionSet to_hex("to_hex");
+	ScalarFunctionSet from_hex("from_hex");
```

Comment:
Could we add the `hex` and `unhex` aliases from [MySQL](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_hex)?